nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PutSlack.java

```diff
@@ -0,0 +1,238 @@
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ICON_URL = new PropertyDescriptor
+            .Builder()
+            .name("icon-url")
+            .displayName("Icon URL")
+            .description("Icon URL to be used for the message")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.URL_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor ICON_EMOJI = new PropertyDescriptor
+            .Builder()
+            .name("icon-emoji")
+            .displayName("Icon Emoji")
+            .description("Icon Emoji to be used for the message. Must begin and end with a colon, e.g. :ghost:")
+            .required(false)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .addValidator(new EmojiValidator())
+            .build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("FlowFiles are routed to success after being successfully sent to Slack")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description("FlowFiles are routed to failure if unable to be sent to Slack")
+            .build();
+
+    public static final List<PropertyDescriptor> descriptors = Collections.unmodifiableList(
+            Arrays.asList(WEBHOOK_URL, WEBHOOK_TEXT, CHANNEL, USERNAME, ICON_URL, ICON_EMOJI));
+
+    public static final Set<Relationship> relationships = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if ( flowFile == null ) {
+            return;
+        }
+
+        try {
+            URL url = new URL(context.getProperty(WEBHOOK_URL).getValue());
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+            conn.setRequestMethod("POST");
+            conn.setDoOutput(true);
+            DataOutputStream outputStream  = new DataOutputStream(conn.getOutputStream());
+
+            JsonObjectBuilder builder = Json.createObjectBuilder();
+            String text = context.getProperty(WEBHOOK_TEXT).evaluateAttributeExpressions(flowFile).getValue();
+            if (text != null && !text.isEmpty()) {
+                builder.add("text", text);
+            } else if (text == null || (text != null && !text.isEmpty())) {
```

Comment:
I think there's a negative logic bug here. The else clause will short-circuit on the text==null check so you can safely assume text != null in the second clause. However then it should check if text is empty, right now it checks to see if it's NOT empty, which would've satisfied the if-clause. So if you have empty text, it will bypass this clause and send an empty body, resulting in a 500 HTTP Response.